### PR TITLE
fix memcpy() crash, add missed cmd in guide, fix softmax

### PR DESCRIPTION
**Pull request by NeoZhangJianyu**
1. When install latest Intel GPU driver, memcpy() from host to device will crash.
fix it by use host buf.
2. Add missed cmd in guide.
3. Fix softmax error in UT.

### CODE REVIEW

**Comment by NeoZhangJianyu** [comment]
@slaren I have tried a solution to return support_mmap by backend in callback function.
But the caller is too earlier that the backend is not initial at that time.
So, I define a static function in ggml-backend.c

**Comment by NeoZhangJianyu** [comment]
@slaren 
Sorry, this is misunderstanding.
memcpy() usage is OK.
due to the new driver issue, the memcpy() will crash.
The workaround is disable mmap() used in sycl backend.

**Comment by slaren** [comment]
If the driver crashes when it receives an address from `mmap`, then what you can do is `memcpy` the buffer to a temporary buffer allocated in a conventional mean, eg. with `malloc` or `std::vector`, and pass that to the driver.

**Comment by NeoZhangJianyu** [comment]
driver is not crash.
The source ptr  provided by mmap is unavailable. That lead to memcpy() crash.

**Comment by slaren** [comment]
memcpy the ptr you get in the `tensor_set` function to a temporary buffer. host -> host. Then memcpy host -> device using the temporary buffer.

**Comment by NeoZhangJianyu** [comment]
host -> host -> device. this solution should be OK.
Let me try.

**Comment by NeoZhangJianyu** [comment]
@slaren Yes, it works well.
I will update the PR as soon.
Thank you!

**Comment by NeoZhangJianyu** [comment]
@slaren @airMeng I update with the new solution. Please review again!

File: ggml-sycl.cpp [hunk 1]
```diff
@@ -16814,11 +16821,13 @@ static void ggml_backend_sycl_buffer_set_tensor(ggml_backend_buffer_t buffer,
     const dpct::queue_ptr stream = g_syclStreams[ctx->device][0];
     SYCL_CHECK(
         CHECK_TRY_ERROR(dpct::dev_mgr::instance().get_device(ctx->device).queues_wait_and_throw()));
-
+    char* host_buf = (char*)malloc(size);
+    memcpy(host_buf, data, size);
```
Comment (airMeng) [inline ggml-sycl.cpp:16825]: Is this allocation necessary only for address from mmap?

Comment (NeoZhangJianyu) [reply ggml-sycl.cpp:16852]: yes. in fact this set_tensor() is only called to copy from mmap address to device after trace.

File: ggml-sycl.cpp [hunk 2]
```diff
@@ -16802,22 +16809,50 @@ catch (sycl::exception const &exc) {
   std::exit(1);
 }
 
+class host_buffer {
```
Comment (airMeng) [inline ggml-sycl.cpp:16812]: maybe a unique ptr?

Comment (airMeng) [inline ggml-sycl.cpp:16812]: approved after this is removed

**Review by airMeng — APPROVED** [review]

File: ggml-sycl.cpp (hunk 2 above)
Comment (NeoZhangJianyu) [reply ggml-sycl.cpp]: yes, remove host_buffer. use malloc/free for thread safe and better performance.

**Comment by zhouwg** [comment]
> driver is not crash. The source ptr provided by mmap is unavailable. That lead to memcpy() crash. If disable mmap() and read the gguf file into host buf. The memcpy() is from host buf to device. It's OK. If memcpy from ptr of mmap() to device, it will crash.

why the source ptr provided by mmap is unavailable? I'd like to reproduce this issue in my local dev, could you provide a quick guide for this: whether I only need to read the docs/backend/SYCL.md?

**Comment by zhouwg** [comment]
your explanation in https://github.com/ggml-org/llama.cpp/pull/12734 is very clear, thanks so much.

**Pull request merged**